TombEngine/Game/control/flipeffect.cpp

```diff
@@ -78,7 +78,8 @@ function<EffectFunction> effect_routines[NUM_FLIPEFFECTS] =
 	MeshSwapToPour,				//43
 	MeshSwapFromPour,			//44
 	LaraLocationPad,			//45
-	KillActiveBaddys			//46
+	KillActiveBaddys,			//46
+	ActiveNextAnimOnLaraSlot	//47
```

Comment:
Why new flipeffect ID was added? There should be empty flipeffect IDs to be reused. Also, what was the initial purpose of adding this flipeffect at all? It seems it has a very narrow usage, and its meaning may be obscure to a builder?

Reply:
Is a problem that I've got. 

Other partners have used animations from ID_LARA_EXTRA_ANIMS, but these were death animations, so they didn't need to return Lara's normal anims. However in my case, I tried get those animations for a custom switch, after this animation ends, I needed to return animation 11 of the ID_LARA. 

This is the example (done with OCB 7 and lua code).
[![TNT like switch](https://img.youtube.com/vi/2FpvW2gjeSc/0.jpg)](https://www.youtube.com/watch?v=2FpvW2gjeSc)

It ocurred me multiples ways to do that return to ID_LARA, but I though that adding the flipeffect could be the less intrussive, and more flexible.

Still my question about this remains, if there were another method in TEN designed to deal with this problem, I would be glad to adapt it.

(Also know that this is not an unique case, eventually if I rework puzzles, the same problem with come with the TR2 gong or the TR1 Midas hand that is used to turn silver pickups into gold puzzles items, and when the others do the dragon, again will have it with the dagger's animation, and the Lara's TR2 wake up animation, plus any new animation that may get added in the future for who knows what?).